### PR TITLE
Enable LLVM toolchain for building LKL targets.

### PR DESCRIPTION
**Pull request by rodionov**
Initial commit enabling clang for LKL to start with fuzzing.

### CODE REVIEW

**Comment by tavip** [comment]
@thehajime any idea why OpenVPN fails to install on the windows target?

**Comment by thehajime** [comment]
> @thehajime any idea why OpenVPN fails to install on the windows target?

```
Microsoft Windows [Version 10.0.17763.3887]
(c) 2018 Microsoft Corporation. All rights reserved.

cat: 'c:\msilog.txt': No such file or directory
```

I have zero idea, sorry.  my best guess is that the runner image of windows was updated last week and has some different behavior on the msi installation.

https://github.com/actions/runner-images/commit/74541db0e609d8dec4cfae620fc8fa67e5a4c6eb

you can delete (or comment out) the particular line on `cat c:\\msilog.txt` from the yaml file.

**Comment by rodionov** [comment]
Just added a CI target for clang-build test.

The checkpatch emits a warning for use of `volatile` keyword in the definition of `unsigned long wrong_size_cmpxchg(volatile void *ptr)` in misc.c. Remove the `volatile` keyword results in the compile time error due to the mismatch of the function prototypes in definition vs declaration. @thehajime @tavip are you OK with this warning or would like to have it ultimately fixed?

File: arch/lkl/kernel/misc.c [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
 
+// Functions __generic_xchg_called_with_bad_pointer and wrong_size_cmpxchg
+// are called in __generic_xchg and __generic_cmpxchg_local respectively.
+// They should be optimized out by the compiler due to the function
+// inlining. However, when building with clang there are some instances
+// where the functions aren't inlined and, thus, the compile-time optimization
+// doesn't eliminate them entirely. As a result, the linker throws
+// unresololved symbols error. The fix below define these functions to
+// bypass the link-time error.
+void __generic_xchg_called_with_bad_pointer(void)
```
Comment (thehajime) [inline arch/lkl/kernel/misc.c:11]: I also saw this issue when I build LKL w/ modified optimization options (e.g., `CFLAGS_page_alloc.o=-O0` in mm/Makefile IIRC).  I couldn't find a proper fix for that then work-around with the similar fix as this.

I'm fine for this fix with the detailed comment.  maybe add TODO tag ?

**Comment by thehajime** [comment]
> @thehajime any idea why OpenVPN fails to install on the windows target?

sorry, it looks like the installation of openvpn isn't triggered at all.  Even commenting out the line of interest, make test failed on windows due to lack of openvpn installation.

please revert the changes on the comment out (the cat msi.log part);  I will investigate it later independent from this PR.

Thanks.

**Review by thehajime — CHANGES_REQUESTED** [review]
LGTM.

Please revert the change as I commented (https://github.com/lkl/linux/pull/512#issuecomment-1406071420), other than that I'm fine with this patchset.

As for the warnings on volatile keyword, I'm fine with that too.

**Comment by thehajime** [comment]
maybe we can close https://github.com/lkl/linux/pull/478 if this is merged.

File: arch/lkl/kernel/misc.c (hunk 1 above)
Comment (rodionov) [reply arch/lkl/kernel/misc.c:11]: Added TODO tag as suggested.

**Comment by rodionov** [comment]
Thank you, @thehajime! I reverted commented out changes in the yaml file.

**Review by thehajime — APPROVED** [review]

**Review by tavip — APPROVED** [review]
LGTM, thanks Eugene!

**Pull request merged**